spark/v3.5/spark-extensions/src/test/java/org/apache/iceberg/spark/extensions/TestRewriteDataFilesProcedure.java

```diff
@@ -109,6 +110,7 @@ public void testRewriteDataFilesOnPartitionTable() {
 
     List<Object[]> actualRecords = currentData();
     assertEquals("Data after compaction should not change", expectedRecords, actualRecords);
+    assertThat(snapshotSummary()).containsKey(CatalogProperties.APP_ID);
```

Comment:
I think it's generally better to just add a separate small test that makes sure app-id / engine-version / engine-name are present in the summary

Reply:
@ajantha-bhat suggested adding validation instead of a separate test not to increase CI time. I'm ok with both ways.

Reply:
I'd vote for having a separate small test that specifically just tests that xyz is in the summary